lib/tasks/helpers/compact_index_tasks_helper.rb

```diff
@@ -0,0 +1,18 @@
+module CompactIndexTasksHelper
+  module_function
+
+  def update_last_checksum(rubygem, task)
+    last_version = rubygem.versions.order(Arel.sql("COALESCE(yanked_at, created_at) desc, number desc, platform desc")).first
```

Comment:
If you move this line out of the helper and pass only the version, you can reuse the helper also for `rake compact_index:correct_info_checksum`.

Reply:
`correct_info_checksum` needs to update the last version as well. If you check the versions query, we are using the same order by there as well.

I will update it to fix [these](https://gist.github.com/sonalkr132/617bb282a61cc03a7c41ad6337e59159), I should be able to use same func to update checksum. IMO, it is important that we keep logic to decide which version needs to be updated in this function so that we don't end updating any unintended version (like I initially did with extra deps task).
info checksum of any version other than *the last* doesn't matter.

Reply:
That makes total sense! :+1: